src/main/java/org/jitsi/jigasi/SipGatewaySession.java

```diff
@@ -509,12 +509,30 @@ public void outgoingCallCreated(CallEvent callEvent)
                     if(roomName != null)
                     {
                         Call call = callEvent.getSourceCall();
-                        call.setData("EXTRA_HEADER_NAME.1",
+                        AtomicInteger headerCount = new AtomicInteger(0);
+                        call.setData(
+                            "EXTRA_HEADER_NAME." + headerCount.addAndGet(1),
                             sipProvider.getAccountID()
                                 .getAccountPropertyString(
                                     JITSI_MEET_ROOM_HEADER_PROPERTY,
                                     "Jitsi-Conference-Room"));
-                        call.setData("EXTRA_HEADER_VALUE.1", roomName);
+                        call.setData(
+                            "EXTRA_HEADER_VALUE." + headerCount.get(),
+                            roomName);
+
+                        // passes all extra headers to the outgoing call
+                        callContext.getExtraHeaders().forEach(
```

Comment:
Can the extra headers map be modified by another thread while we iterate over them? I see we return the raw object. Would it make sense to return a read only map?

Reply:
I'm good with the code as it is, maybe you could just add your comment as an XXX in the code so that we can remember this reasoning if and when we decide to touch this again.